utils/ChromiumDownloader.js

```diff
@@ -193,7 +207,7 @@ function downloadFile(url, destinationPath, progressCallback) {
   const promise = new Promise((x, y) => { fulfill = x; reject = y; });
 
   const options = requestOptions(url);
-  const request = https.get(options, response => {
+  const request = protocolSelector(url).get(options, response => {
```

Comment:
Let's rename `requestOptions(url, method)` into `httpRequest(url, method)` and handle the http/https selection inside.

So here instead of getting options and then initiating a request, we'll have just

```js
const request = httpRequest(url, 'GET', response => {
  // ..
});
```

Reply:
That's a nice solution, now that I see it. I'll update the code!

Reply:
I digged a bit down in the code, and it seems that you are handling proxies with the following modules: `proxy-from-env` & `https-proxy-agent`.

This is needed for proxy support, but my suggestion would just be to replace it all with [request](https://www.npmjs.com/package/request), this has all the features while being simple and would greatly simplify the code needed. This would handle https/http/http2, proxies ect. It would also allow for the use of promises and Async/Await.

How would you feel about this?

Reply:
@Saturate `request` has a huge dependency tree. We intentionally avoid using it since the functionality is only needed for a single file download.

Reply:
How about http://npm.anvaka.com/#/view/2d/axios

Reply:
Didn't really dig much into `request` but, I made a mistake thinking it was good as `npm` used it. I did try to implement the download with it, and it works fine with all proxy settings ect. However I can see that it's huge for the little things that we would use it for.

As far as I can see the `axios` packages does not support proxies via env vars, so we won't get that for free, however it does seem to figure out switching between HTTP and HTTPS. I was thinking https://www.npmjs.com/package/needle but they seem pretty similar, however `needle` has very few dependencies, even less than axios it would seem.

I could implement it with either pretty easily. Keeping the following dependencies for handling proxies:
```
const ProxyAgent = require('https-proxy-agent');
const getProxyForUrl = require('proxy-from-env').getProxyForUrl;
```

On a side note, I found that the `node .\check_availability.js` script right now does not handle proxies, and thus does not work for me. I'll rewrite that as well.

Reply:
I'll just do what @aslushnikov proposed 5 days ago, and I'll make sure that it supports proxies, and reads the proxies the same way NPM does, this would be the minimal change and we would not have to add any new dependencies.

Reply:
> Would a axios version be welcome?

@Saturate we've spent quite a bit of time figuring the `https-proxy-agent` and `proxy-from-env` dependencies a while ago. We're pretty happy with the results so far, let's keep them as-is and avoid adding anything new.